packages/general/src/util/Mutex.ts

```diff
@@ -65,6 +65,21 @@ export class Mutex implements PromiseLike<unknown> {
         }
     }
 
+    /**
+     * Enqueue work with an awaitable result.
+     */
+    produce<T>(task: () => PromiseLike<T>, cancel?: () => void): Promise<T> {
```

Comment:
Should we then kill PromiseQueue? ;-) because was/is exactly this :-)

Reply:
Hmm, I was thinking intent of PromiseQueue is for it support multiple workers...  So not a mutex but a work queue.  Could maybe have a shared implementation since a work queue with one worker is effectively a mutex but I think it makes sense to have both primitives

Reply:
Ahh ok , yes makes sense. then fine!